packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -55,6 +55,7 @@ export default util.createRule({
     return {
       [[
         'LogicalExpression[operator="&&"] > Identifier',
+        'LogicalExpression[operator="&&"] > MemberExpression',
         'LogicalExpression[operator="&&"] > BinaryExpression[operator="!=="]',
         'LogicalExpression[operator="&&"] > BinaryExpression[operator="!="]',
       ].join(',')](
```

Comment:
You need to update the node type with this selector change as well

Reply:
@bradzacher Oops :) Thanks!